malariagen_data/ag3.py

```diff
@@ -100,34 +100,54 @@ def releases(self):
                 self._cache_releases = public_releases
         return self._cache_releases
 
-    def sample_sets(self, release="v3"):
-        """Access the manifest of sample sets.
+    def sample_sets(self, release=None):
+        """Access a dataframe of sample sets.
 
         Parameters
         ----------
-        release : str
+        release : str, optional
             Release identifier. Give "v3" to access the Ag1000G phase 3 data release.
 
         Returns
         -------
         df : pandas.DataFrame
+            A dataframe of sample sets, one row per sample set.
 
         """
 
-        if release not in self.releases:
-            raise ValueError(f"Release not available: {release!r}")
+        if release is None:
+            # retrieve sample sets from all available releases
+            release = self.releases
 
-        try:
-            return self._cache_sample_sets[release]
+        if isinstance(release, str):
+            # retrieve sample sets for a single release
 
-        except KeyError:
-            path = f"{self._path}/{release}/manifest.tsv"
-            with self._fs.open(path) as f:
-                df = pandas.read_csv(f, sep="\t", na_values="")
-            df["release"] = release
-            self._cache_sample_sets[release] = df
+            if release not in self.releases:
+                raise ValueError(f"Release not available: {release!r}")
+
+            try:
+                return self._cache_sample_sets[release]
+
+            except KeyError:
+                path = f"{self._path}/{release}/manifest.tsv"
+                with self._fs.open(path) as f:
+                    df = pandas.read_csv(f, sep="\t", na_values="")
+                df["release"] = release
+                self._cache_sample_sets[release] = df
+                return df
+
+        elif isinstance(release, (list, tuple)):
+            # retrieve sample sets from multiple releases
+            df = pandas.concat(
+                [self.sample_sets(release=r) for r in release],
+                axis=0,
+                ignore_index=True,
```

Comment:
can we cache here?

Reply:
I don't think it's worth it, as the dataframes for each of the releases in the list will be cached.